apps/labs/posts/nebari-typer-cli.md

```diff
@@ -0,0 +1,162 @@
+---
+title: 'The evolution of Nebari CLI using Typer'
+published: October 4, 2022
+author: asmi-jafar
+description: 'Nebari typer CLI'
+featuredImage:
+  src: /posts/asmi-blogposts/intern-mentors.png
+  alt: 'An illustration of nebari logo with the assigned mentors and intern.'
+hero:
+  imageSrc: /posts/asmi-blogposts/typer-nebari.png
+  imageAlt: 'Nebari logo with Typer CLI logo'
+---
+
+Hey, I'm [Asmi Jafar](https://twitter.com/asmijafar20), a computer science postgraduate student from India.
+I've worked on the [Nebari project](https://www.nebari.dev/) (previously known as Qhub) during my three months internship at Quansight Labs.
+
+## 🪴 About Nebari
+
+Nebari is an open source project from Quansight that enables teams/users to deploy a cost-effective and scalable compute/data science platforms in the cloud. It uses an infrastructure-as-a-code approach to quickly and smoothly deploy a shared data science environment, on-premises or in the cloud of choice. For more details, refer to the [Nebari docs](https://www.nebari.dev/).
+
+## Nebari Command-line Interface (CLI)
+
+I've worked on developing the Nebari [Command-line interface (CLI)](https://en.wikipedia.org/wiki/Command-line_interface) during my three months internship.
+
+The Nebari CLI consists of various commands the user needs to run to deploy, configure, and update Nebari. Let's see briefly what each one of the `nebari` commands does.
+
+Below is a diagram of the Nebari CLI commands.
+
+<img alt="Nebari commands" src="/posts/asmi-blogposts/nebari-commands.png" />
+
+1. `nebari init` command: initializes and creates the `nebari-config.yaml` file for the deployment.
+2. `nebari validate` command: validates the configuration file `nebari-config.yaml`.
+3. `nebari render` command: renders the infrastructure configuration files (all terraform stages) and resources.
+4. `nebari deploy` command: deploys the Nebari with all the resources and configurations set in `nebari-config.yaml` file.
+5. `nebari destroy` command: tears down the provisioned resources for Nebari (the ones which were created during deployment).
+
+## ✅ Initial steps
+
+Prior to my internship, Nebari already has a command line interface with all the functionality listed above however it was clunky and, at times, a little challenging to use. My task as an intern was to improve the CLI experience for the end-user. After exploring various tools and libraries it was decided that we would proceed with [Typer](https://typer.tiangolo.com/) and [Rich](https://rich.readthedocs.io/en/stable/introduction.html). For those unfamiliar, Typer is a Python library that makes CLI applications more interactive and user friendly. I started exploring features of `typer` and how it could be implemented as a replacement for the current CLI. After exploring about typer and rich, Now it is time to implement `typer` on the predefined functions of the CLI.
```

Comment:
```suggestion
Prior to my internship, Nebari had an [`argparse`-based](https://docs.python.org/3/library/argparse.html) command line interface which included all the functionality listed above. However, it was clunky and at times, a little challenging to use. My task as an intern was to improve the CLI experience for the end-user. After exploring various tools and libraries it was decided that we would proceed with [Typer](https://typer.tiangolo.com/) and [Rich](https://rich.readthedocs.io/en/stable/introduction.html). For those unfamiliar, Typer is a Python library that makes CLI applications more interactive and user friendly. I started exploring features of `typer` and how it could be implemented as a replacement for the current CLI.
```